internal/pkg/vm/state/state.go

```diff
@@ -0,0 +1,204 @@
+package state
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/encoding"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/actor"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-hamt-ipld"
+	cbor "github.com/ipfs/go-ipld-cbor"
+	"github.com/pkg/errors"
+)
+
+// Review: can we get rid of this?
+type Tree interface {
```

Comment:
The interface may be handy to keep, but could probably be declared in consumers (Go style) rather than here (Java style). Your call.

However,  the term "state tree" is a widely used and useful term for the top level of the state. Assuming you delete or move this interface elsewhere, I suggest renaming `State` below to `Tree`, which is a much more descriptive name to people who carry the state tree idea in their head already. `State` is already overloaded.